frontend/src/views/omni/Modals/DownloadTalosctl.vue

```diff
@@ -27,46 +29,44 @@ const close = () => {
   router.go(-1)
 }
 
-onMounted(async () => {
-  // Its sad that you cannot execute async function in computed
-  const url = '/talosctl/downloads'
-
+const talosctlRelease = computedAsync(async () => {
   try {
-    const res = await fetch(url)
-    talosctlRelease.value = await res.json()
+    const res = await fetch('/talosctl/downloads')
+    return (await res.json()) as ResponseData
   } catch (e) {
     showError(e.message)
-    return
   }
+})
 
-  defaultVersion.value = talosctlRelease.value?.release_data.default_version
-  selectedVersion.value = defaultVersion.value
-
-  defaultValue.value =
+const availableVersions = computed(
+  () =>
+    new Map(
+      Object.entries(talosctlRelease.value?.release_data.available_versions ?? {}).sort(
+        ([a], [b]) => compare(a.replace('v', ''), b.replace('v', '')),
+      ),
+    ),
+)
+
+const defaultVersion = computed(() => Array.from(availableVersions.value.keys()).pop())
+const defaultPlatform = computed(
+  () =>
     defaultVersion.value &&
-    talosctlRelease.value?.release_data.available_versions[defaultVersion.value].find((item) =>
-      item.url.endsWith('linux-amd64'),
-    )?.name
-
-  selectedOption.value = defaultValue.value
-})
+    availableVersions.value
+      .get(defaultVersion.value)
+      ?.find((item) => item.url.endsWith('linux-amd64'))?.name,
```

Comment:
It could be cool if we could determine the OS and pick the right one automatically :)